RadixWallet/Features/AccountPreferencesFeature/Children/UpdateAccountLabel+View.swift

```diff
@@ -31,51 +32,78 @@ extension UpdateAccountLabel {
 		let sanitizedName: NonEmptyString?
 		let updateButtonControlState: ControlState
 		let hint: Hint?
+		let textFieldFocused: Bool
 	}
 
 	@MainActor
 	public struct View: SwiftUI.View {
 		let store: StoreOf<UpdateAccountLabel>
+		@Environment(\.dismiss) var dismiss
+		@FocusState private var textFieldFocus: Bool
 
 		init(store: StoreOf<UpdateAccountLabel>) {
 			self.store = store
 		}
 
 		public var body: some SwiftUI.View {
+			content
+				.withNavigationBar {
+					dismiss()
+				}
+				.presentationDetents([.fraction(0.5)])
+				.presentationDragIndicator(.visible)
+				.presentationBackground(.blur)
+		}
+
+		private var content: some SwiftUI.View {
 			WithViewStore(store, observe: \.viewState, send: { .view($0) }) { viewStore in
-				VStack {
-					VStack(alignment: .center, spacing: .medium1) {
+				VStack(spacing: .zero) {
+					VStack(spacing: .medium1) {
+						Text(L10n.AccountSettings.RenameAccount.title)
+							.textStyle(.sheetTitle)
+							.multilineTextAlignment(.center)
+
+						Text(L10n.AccountSettings.RenameAccount.subtitle)
+							.textStyle(.body1Regular)
+							.multilineTextAlignment(.center)
+
 						let nameBinding = viewStore.binding(
 							get: \.accountLabel,
 							send: { .accountLabelChanged($0) }
 						)
 						AppTextField(
-							primaryHeading: .init(text: L10n.AccountSettings.RenameAccount.subtitle),
 							placeholder: "",
 							text: nameBinding,
-							hint: viewStore.hint
+							hint: viewStore.hint,
+							focus: .on(
+								true,
+								binding: viewStore.binding(
+									get: \.textFieldFocused,
+									send: { .focusChanged($0) }
+								),
+								to: $textFieldFocus
+							)
 						)
 						.keyboardType(.asciiCapable)
 						.autocorrectionDisabled()
+					}
+					.foregroundColor(.app.gray1)
+					.padding(.horizontal, .medium3)
 
-						WithControlRequirements(
-							viewStore.sanitizedName,
-							forAction: { viewStore.send(.updateTapped($0)) }
-						) { action in
-							Button(L10n.AccountSettings.SpecificAssetsDeposits.update) {
-								action()
-							}
+					Spacer()
+
+					WithControlRequirements(
+						viewStore.sanitizedName,
+						forAction: { viewStore.send(.updateTapped($0)) }
+					) { action in
+						Button(L10n.AccountSettings.SpecificAssetsDeposits.update, action: action)
```

Comment:
I think there should be a separator above the button (consider using `.footer`). See this [zeplin](https://app.zeplin.io/project/639c7e7ab391bb15ac5d25fa/screen/661705eb178b448f466c508e) screen as a reference.